node/app/db_console.go

```diff
@@ -808,11 +803,46 @@ func (c *DBConsole) Run() {
 			grpc.MaxCallRecvMsgSize(600*1024*1024),
 		),
 	)
+}
+
+type TokenBalance struct {
+	Owned            *big.Int
+	UnconfirmedOwned *big.Int
+}
+
+func FetchTokenBalance(client protobufs.NodeServiceClient) (TokenBalance, error) {
+	info, err := client.GetTokenInfo(
+		context.Background(),
+		&protobufs.GetTokenInfoRequest{},
+	)
 	if err != nil {
-		panic(err)
+		return TokenBalance{}, errors.Wrap(err, "error getting token info")
 	}
 
+	conversionFactor, _ := new(big.Int).SetString("1DCD65000", 16)
+
+	owned := new(big.Int).SetBytes(info.OwnedTokens)
+	owned.Div(owned, conversionFactor)
+
+	unconfirmedOwned := new(big.Int).SetBytes(info.UnconfirmedOwnedTokens)
+	unconfirmedOwned.Div(unconfirmedOwned, conversionFactor)
+
+	return TokenBalance{
+		Owned:            owned,
+		UnconfirmedOwned: unconfirmedOwned,
+	}, nil
+}
+
+// Runs the DB console
+func (c *DBConsole) Run() {
+	conn, err := ConnectToNode(c.nodeConfig)
+	if err != nil {
+		panic(err)
+	}
 	defer conn.Close()
+
+	grpcWarn := conn.Target() == defaultGrpcAddress
```

Comment:
This check is not equivalent to the prior version if the user's grpc address in the config is `localhost:8337`. Let me know please if I should change it to match the prior logic exactly

Reply:
Alternative could be to do this:

```
grpcWarn := c.nodeConfig.ListenGRPCMultiaddr != ""
```

This wouldn't match the prior logic either exactly since it doesn't check if the config value is correct, but maybe it's closer to the original intention which I don't fully understand tbh 😅

Reply:
Ok I think I got it now: if `grpcWarn` is true, we want to display the warning `"gRPC Not Enabled, Please Configure"` and otherwise show the connection status message.

https://github.com/QuilibriumNetwork/ceremonyclient/blob/7a113ebb785907eed266a6f126f4f6eecc9cb82f/node/app/db_console.go#L464-L478

So the purpose is to warn the user if they haven't set the necessary config field yet. We can detect this if `nodeConfig.ListenGRPCMultiaddr == ""`.

I fixed the db console in [7a113eb](https://github.com/QuilibriumNetwork/ceremonyclient/pull/38/commits/7a113ebb785907eed266a6f126f4f6eecc9cb82f) and added the same warning to the CLI in [bac0344](https://github.com/QuilibriumNetwork/ceremonyclient/pull/38/commits/bac034450ce585e7528a44820053ea0eb8e3d23f).

I retested the `-balance` flag and the balance in the DB console on a node with gRPC enabled. I also tested that the warnings are shown for a node with gRPC disabled in both the CLI and the DB console.